core/snmp/proxy-rpc-impl/src/main/java/org/opennms/netmgt/snmp/proxy/common/SnmpProxyRpcModule.java

```diff
@@ -176,6 +181,10 @@ private static final void addResult(SnmpResult result, String correlationId, Map
         final CompletableFuture<SnmpValue[]> future = SnmpUtils.getAsync(request.getAgent(), oids);
         return future.thenApply(values -> {
             final List<SnmpResult> results = new ArrayList<>(oids.length);
+            if (oids.length != values.length) {
+                LOG.error("Response values size {} is not matching request oids size {}", values.length, oids.length);
+                throw new SnmpResponseException("Invalid SNMP Response, Number of Response elements are not matching number of Requested oids");
```

Comment:
I think better behavior would be to return the results we did retrieve rather than none at all.